e2e/cypress/integration/messaging/markdown_preview_inline_image_spec.js

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Messaging', () => {
+    before(() => {
+        // # Login, set the Show Markdown Preview preference, then go to /
+        cy.apiLogin('user-1');
+        cy.apiSaveShowMarkdownPreviewPreference();
+        cy.visit('/');
+    });
+
+    it('M18714-Markdown preview: inline image', () => {
+        const message = '![make it so](https://i.stack.imgur.com/MNeE7.jpg)';
+
+        cy.visit('/ad-1/channels/town-square');
+
+        // # Get the height before starting to write
+        cy.get('#post_textbox').should('be.visible').clear().then(() => {
+            cy.get('#post-create').then((postArea) => {
+                cy.wrap(parseInt(postArea[0].clientHeight, 10)).as('initialHeight');
+            });
+        });
+
+        // # Post first line to use
+        cy.get('#post_textbox').type(message).type('{shift}{enter}').type(message);
+
+        // # Click on Preview button
+        cy.get('#previewLink').click({force: true});
+
+        // * Image does not overlap Post List
+        cy.get('#post-list').then((postList) => {
+            cy.get('.markdown-inline-img').then((img) => {
+                expect(postList[0].getBoundingClientRect().bottom).lessThan(img[0].getBoundingClientRect().top);
+                expect(postList[0].getBoundingClientRect().bottom).lessThan(img[1].getBoundingClientRect().top);
+                expect(img[0].getBoundingClientRect().bottom <= img[1].getBoundingClientRect().top || img[0].getBoundingClientRect().right <= img[1].getBoundingClientRect().left).equals(true);
```

Comment:
I don't understand what is this for.  Feels wrong to me 'cause the right side of first image will always be greater than the left side of second image.
```
img[0].getBoundingClientRect().right <= img[1].getBoundingClientRect().left
```

Please split the test in `||` if necessary.

Also, please add comments on what the test is all about.

Reply:
I suppose (maybe I am wrong) that the second image, if it is small enough and have enough space, will be next to each other, instead of one on the top of the other.  If that is the case, we have to do the check in horizontal. Therefore, this check can be read as "images do not overlap on vertical, or they do not overlap in horizontal". If that can never happen, this test can be removed.

Reply:
Oh I see, thanks for clarifying.  I think we can remove it since the inline image is same and the window size is defined in Cypress.  So, it will not change and will not cause flaky test.

Reply:
Done on fc6f185